src/variant_object.cpp

```diff
@@ -164,6 +164,21 @@ namespace fc
    }
 
 
+   bool operator ==( const variant_object& obj1, const variant_object& obj2 ) {
```

Comment:
the order which these iterators traverse a `(mutable_)?variant_object` is not strictly defined as ordered.  As a result this comparison will return `false` when comparing these two objects:

```
auto lhs = mutable_variant_object()("foo", 1)("bar", 1);
auto rhs = mutable_variant_object()("bar", 1)("foo", 1);

// expect are_equal to be true but it is not!
bool are_equal = (lhs == rhs); 
```